src/plugins.c

```diff
@@ -2067,8 +2055,14 @@ gboolean geany_plugin_register_proxy(GeanyPlugin *plugin, const gchar **extensio
 
 	foreach_strv(ext, extensions)
 	{
+		if (**ext == '.')
+		{
+			g_warning(_("Proxy plugin '%s' extension '%s' starts with a dot. "
+				"Please fix your proxy plugin."), p->info.name, *ext);
+		}
```

Comment:
Returning false is a bit involved, since you'd want to roll back adding earlier extensions from the same call.